benchmarks/src/main/kotlin/chat/BenchmarkConfiguration.kt

```diff
@@ -0,0 +1,192 @@
+
+/**
+ * Execution time of each benchmark.
+ */
+const val BENCHMARK_TIME_MS = 2000L
+/**
+ * Warm up iterations count
+ */
+const val WARM_UP_ITERATIONS = 5
+/**
+ * Benchmark iterations count
+ */
+const val BENCHMARK_ITERATIONS = 5
+/**
+ * CSV file containing the configurations and final metrics of the executed benchmarks
+ */
+const val BENCHMARK_OUTPUT_FILE = "results.csv"
+/**
+ * Folder containing all benchmark output files
+ */
+const val BENCHMARK_OUTPUT_FOLDER = "src/main/kotlin/chat/benchmarkresults/"
+/**
+ * The header for the output CSV file. It should be coordinated with [chat.BenchmarkConfiguration.toCSV] method.
+ */
+const val BENCHMARK_CONFIGURATION_FIELDS = "threads,userCount,maxFriendsPercentage,channelType,averageWork,benchmarkMode,dispatcherType,avgSentMessages,stdSentMessages,avgReceivedMessages,stdReceivedMessages"
+
+/**
+ * Class containing all the benchmark configuration and metrics of the run benchmark
+ */
+class BenchmarkConfiguration(
+        var threads: Int,
+        var users: Int,
+        var maxFriendsPercentage: Double,
+        var channelType: ChannelType,
+        var averageWork: Int,
+        var benchmarkMode: BenchmarkModes,
+        var dispatcherType: DispatcherTypes
+) {
+    val sentMessagesPerRun = ArrayList<Double>()
+
+    val receivedMessagesPerRun = ArrayList<Double>()
+
+    override fun toString(): String {
+        return "BenchmarkConfiguration(threads=$threads, users=$users, maxFriendsPercentage=$maxFriendsPercentage, channelType=$channelType, averageWork=$averageWork, benchmarkMode=$benchmarkMode, dispatcherType=$dispatcherType, sentMessagesPerRun=$sentMessagesPerRun, receivedMessagesPerRun=$receivedMessagesPerRun)"
+    }
+
+    fun configurationToString() : String {
+        return "[threads=$threads, users=$users, maxFriendsPercentage=$maxFriendsPercentage, channelType=$channelType, averageWork=$averageWork, benchmarkMode=$benchmarkMode, dispatcherType=$dispatcherType]"
+    }
+
+    fun toCSV(): String {
+        val standardDeviation = StandardDeviation()
+        val avgSentMessages = sentMessagesPerRun.average()
+        val avgReceivedMessages = receivedMessagesPerRun.average()
+        val stdSentMessages = standardDeviation.evaluate(sentMessagesPerRun.toDoubleArray())
+        val stdReceivedMessages = standardDeviation.evaluate(receivedMessagesPerRun.toDoubleArray())
+        return "$threads,$users,$maxFriendsPercentage,$channelType,$averageWork,$benchmarkMode,$dispatcherType,%.2f,%.2f,%.2f,%.2f".format(avgSentMessages, stdSentMessages, avgReceivedMessages, stdReceivedMessages)
+    }
+
+    fun toArray() : Array<String> {
+        return arrayOf(threads.toString(), users.toString(), maxFriendsPercentage.toString(), channelType.toString(), averageWork.toString(), benchmarkMode.toString(), dispatcherType.toString())
+    }
+
+    companion object {
+        fun parseArray(array : Array<String>) : BenchmarkConfiguration {
```

Comment:
Why do you need this?

Reply:
This benchmark spawns a new process, and passes benchmark configuration to this process. Created process parse configurations.